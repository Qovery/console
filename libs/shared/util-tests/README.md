# shared-util-tests

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-tests` to execute the unit tests via [Jest](https://jestjs.io).
