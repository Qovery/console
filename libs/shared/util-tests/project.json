{
  "name": "shared-util-tests",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util-tests/src",
  "projectType": "library",
  "tags": ["scope:shared", "type:util"],
  "implicitDependencies": ["!shared-ui"],
  "targets": {
    "test": {
      "options": {
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "coverage": true
        }
      }
    }
  }
}
