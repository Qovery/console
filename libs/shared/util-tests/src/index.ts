export * from './lib/shared-util-tests'
