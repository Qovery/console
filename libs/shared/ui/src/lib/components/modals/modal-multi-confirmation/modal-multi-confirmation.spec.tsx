import { act, fireEvent, renderWithProviders, screen } from '@qovery/shared/util-tests'
import ModalMultiConfirmation, { type ModalMultiConfirmationProps } from './modal-multi-confirmation'

describe('ModalMultiConfirmation', () => {
  let props: ModalMultiConfirmationProps

  beforeEach(() => {
    props = {
      title: 'my title',
      description: 'my description',
      checks: ['one', 'two'],
      callback: jest.fn(),
    }
  })

  it('should render successfully', () => {
    const { baseElement } = renderWithProviders(<ModalMultiConfirmation {...props} />)
    expect(baseElement).toBeTruthy()
  })

  it('should render the validation checkboxes', () => {
    renderWithProviders(<ModalMultiConfirmation {...props} />)

    expect(screen.getByText('one')).toBeInTheDocument()
    expect(screen.getByText('two')).toBeInTheDocument()
  })

  it('should not submit until checkboxes are checked', async () => {
    renderWithProviders(<ModalMultiConfirmation {...props} />)

    expect(screen.getByText('one')).toBeInTheDocument()
    expect(screen.getByText('two')).toBeInTheDocument()

    // Checking the first confirmation checkbox
    await act(() => {
      fireEvent.click(screen.getByText('one'))
    })

    // Making sure the submit button is disabled
    fireEvent.click(screen.getByText('Confirm'))
    expect(screen.getByText('Confirm')).toBeDisabled()
  })

  it('should submit if all checkboxes are checked', async () => {
    renderWithProviders(<ModalMultiConfirmation {...props} />)

    // Checking the confirmation checkboxes
    await act(() => {
      fireEvent.click(screen.getByText('one'))
      fireEvent.click(screen.getByText('two'))
    })

    fireEvent.click(screen.getByText('Confirm'))

    // Making sure the submit button is enabled
    expect(screen.getByText('Confirm')).toBeEnabled()
  })

  it('should match delete mode without description', () => {
    props.description = undefined
    props.isDelete = true

    const { container } = renderWithProviders(<ModalMultiConfirmation {...props} />)

    expect(container).toMatchSnapshot()
  })
})
