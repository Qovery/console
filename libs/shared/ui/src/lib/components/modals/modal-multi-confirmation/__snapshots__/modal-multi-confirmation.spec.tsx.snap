// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalMultiConfirmation should match delete mode without description 1`] = `
<div>
  <div
    class="p-6"
  >
    <h2
      class="h4 mb-2 max-w-sm text-neutral-400 dark:text-neutral-50"
    >
      my title
    </h2>
    <div
      class="mb-6 text-sm text-neutral-350 dark:text-neutral-50"
    />
    <form>
      <div
        class="flex flex-row gap-x-3 p-3 border rounded text-sm border-red-500 bg-red-50 text-red-500 mb-5"
        data-accent-color="red"
      >
        <div
          class="-order-2 text-lg leading-6"
        >
          <i
            aria-hidden="true"
            class="fa-solid fa-triangle-exclamation text-red-600"
          />
        </div>
        <div
          class="-order-1 mr-auto gap-x-3 text-neutral-400"
        >
          <div
            class="flex flex-col gap-3"
          >
            <p
              class="font-semibold"
            />
            <div
              class="flex items-center gap-4"
            >
              <div
                class="flex flex-col gap-2"
              >
                <div
                  class="flex items-center gap-3"
                >
                  <button
                    aria-checked="false"
                    class="group transition bg-neutral-100 dark:bg-neutral-500 border border-neutral-300 dark:border-neutral-350 rounded-sm w-[18px] h-[18px] hover:border-2 data-[state=checked]:border data-[state=indeterminate]:border-2 disabled:border disabled:border-neutral-250 hover:border-red-500 data-[state=checked]:bg-red-500 data-[state=checked]:border-red-500 data-[state=indeterminate]:border-red-500 disabled:data-[state=checked]:border-red-300 disabled:data-[state=checked]:bg-red-300 disabled:data-[state=indeterminate]:border-red-300 shrink-0"
                    data-state="unchecked"
                    id="one"
                    role="checkbox"
                    type="button"
                    value="on"
                  />
                  <input
                    aria-hidden="true"
                    name="one"
                    style="transform: translateX(-100%); position: absolute; pointer-events: none; opacity: 0; margin: 0px; width: 0px; height: 0px;"
                    tabindex="-1"
                    type="checkbox"
                    value="on"
                  />
                  <label
                    for="one"
                  >
                    one
                  </label>
                </div>
              </div>
            </div>
            <div
              class="flex items-center gap-4"
            >
              <div
                class="flex flex-col gap-2"
              >
                <div
                  class="flex items-center gap-3"
                >
                  <button
                    aria-checked="false"
                    class="group transition bg-neutral-100 dark:bg-neutral-500 border border-neutral-300 dark:border-neutral-350 rounded-sm w-[18px] h-[18px] hover:border-2 data-[state=checked]:border data-[state=indeterminate]:border-2 disabled:border disabled:border-neutral-250 hover:border-red-500 data-[state=checked]:bg-red-500 data-[state=checked]:border-red-500 data-[state=indeterminate]:border-red-500 disabled:data-[state=checked]:border-red-300 disabled:data-[state=checked]:bg-red-300 disabled:data-[state=indeterminate]:border-red-300 shrink-0"
                    data-state="unchecked"
                    id="two"
                    role="checkbox"
                    type="button"
                    value="on"
                  />
                  <input
                    aria-hidden="true"
                    name="two"
                    style="transform: translateX(-100%); position: absolute; pointer-events: none; opacity: 0; margin: 0px; width: 0px; height: 0px;"
                    tabindex="-1"
                    type="checkbox"
                    value="on"
                  />
                  <label
                    for="two"
                  >
                    two
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex justify-end gap-3"
      >
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 hover:[&:not(:focus-visible):not(:disabled)]:shadow-sm active:[&:not(:focus-visible)]:shadow-none outline-brand-500 text-sm h-11 px-5 rounded border border-transparent text-neutral-350 hover:[&:not(:active):not(:focus-visible)]:border-neutral-300 hover:[&:not(:disabled)]:text-neutral-400 dark:hover:[&:not(:disabled)]:text-neutral-250 focus-visible:text-neutral-400 active:bg-neutral-150 dark:active:bg-transparent active:text-neutral-400 disabled:text-neutral-300 disabled:bg-neutral-150 disabled:border-none"
          type="button"
        >
          Cancel
        </button>
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-red-500 text-sm h-11 px-5 rounded bg-red-500 active:bg-red-600 hover:bg-red-400 text-white disabled:text-red-300 disabled:bg-red-100"
          disabled=""
          type="submit"
        >
          Confirm
        </button>
      </div>
    </form>
  </div>
</div>
`;
